src/Lorentz/Contracts/RegistryDAO/Types.hs

```diff
@@ -0,0 +1,105 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | RegistryDAO Types
+module Lorentz.Contracts.RegistryDAO.Types
+  ( RegistryDaoContractExtra (..)
+  , RegistryEntry (..)
+  , RegistryUpdate (..)
+  , RegistryDaoProposalMetadata (..)
+  , IsoRegistryDaoProposalMetadata
+  ) where
+
+import Lorentz
+
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+
+
+data RegistryDaoContractExtra k v = RegistryDaoContractExtra
+  { ceRegistry :: BigMap k (RegistryEntry v)
+  }
+  deriving stock (Generic)
+
+instance (NiceComparable k, Ord k, IsoValue k, IsoValue v)
+  => (IsoValue (RegistryDaoContractExtra k v))
+
+instance (HasAnnotation k, HasAnnotation v)
+  => HasAnnotation (RegistryDaoContractExtra k v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (Ord k, TypeHasDoc k, TypeHasDoc v)
+  => TypeHasDoc (RegistryDaoContractExtra k v) where
+    typeDocMdDescription =
+      "TODO"
+    typeDocMdReference = poly2TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(RegistryDaoContractExtra ByteString ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(RegistryDaoContractExtra ByteString ByteString)
+
+instance (Ord k) => Default (RegistryDaoContractExtra k v) where
+  def = RegistryDaoContractExtra mempty
```

Comment:
You can avoid `Ord k` constraint if use `def`. `mempty` requires `Ord` only because Monoid also has `<>` where `Ord` is actually necessary.